packages/sanity/src/core/preview/utils/getPreviewStateObservable.ts

```diff
@@ -94,13 +109,25 @@ export function getPreviewStateObservable(
   )
 
   return combineLatest([draft$, published$, version$, versions$]).pipe(
-    map(([draft, published, version, versions]) => ({
-      draft: draft.snapshot ? {title, ...(draft.snapshot || {})} : null,
-      isLoading: false,
-      published: published.snapshot ? {title, ...(published.snapshot || {})} : null,
-      version: version.snapshot ? {title, ...(version.snapshot || {})} : null,
-      versions,
-    })),
+    map(([draft, published, version, versions]) => {
+      let v = version.snapshot ? {title, ...(version.snapshot || {})} : null
+
+      /**
+       * when the search type should be raw, there are versions and the document provided is a version
+       * then we need to find the exact version to display
+       */
+      if (perspective.isRaw && versions && isVersionId(documentId)) {
+        v = {title, ...versions[getVersionFromId(documentId) ?? '']?.snapshot}
+      }
+
+      return {
+        draft: draft.snapshot ? {title, ...(draft.snapshot || {})} : null,
+        isLoading: false,
+        published: published.snapshot ? {title, ...(published.snapshot || {})} : null,
+        version: v,
```

Comment:
I've removed `version` in #7828 because I feel it's very context dependent and I'd prefer to keep it a separate concern from loading what to fetch for previewing. No action required from you, just an FYI.

Reply:
I'll integrate this work into my branch when this lands